gitops/generate.go

```diff
@@ -90,6 +91,23 @@ func Generate(fs afero.Afero, gitOpsFolder string, outputFolder string, componen
 
 // GenerateOverlays generates the overlays dir from a given BindingComponent
 func GenerateOverlays(fs afero.Afero, gitOpsFolder string, outputFolder string, component appstudioshared.BindingComponent, imageName, namespace string, componentGeneratedResources map[string][]string) error {
+	kustomizeFileExist, err := fs.Exists(filepath.Join(outputFolder, kustomizeFileName))
```

Comment:
just do a quick search and see if you can hit the error cases using a different fs? Probably just check the module tests to see if we can do it as well

Reply:
took a quick look, a different fs does not return any error, it will only return error from [os.stat](https://cs.opensource.google/go/go/+/refs/tags/go1.18.4:src/os/stat_unix.go;l=20)
and only if `File` is nil or did not setup correctly. In our code, we do not touch down to lowlevel `File` object. 
and `GenerateOverlays` has no test for it, it's being tested as part of `GenerateAndPushOverlays` test, if the path is invalid, it will be found out and returned before it reaches to this point

Reply:
Should we have a test case for GenerateOverlays then? Would be good to have a test case that validates the contents of the overlays we're generating (similar to `TestGenerateParentKustomize` in https://github.com/redhat-appstudio/application-service/blob/729bfb4664b9a6cd5821a90c65fc655176fd5b87/gitops/generate_test.go#L575)

Reply:
I've added a `TestGenerateOverlays` with unittests. I'm not able to cover the `os.stat` failure case tho, but it should cover other unreached lines and increase coverage.

Reply:
It'll be tricky to test for os.stat failures, so I think this is fine.